### ISSUE TITLE
Lost my world on Fool Craft

### ISSUE THREAD

**Issue by EroAxe** [scalda/FoolCraft#561]
## SUGGESTION/HELP
- [ ] I would like to propose an additional mod, config change, custom recipe, etc.
- [ ] I am requesting help with installing or running the pack

## Issue Description 
I have completely lost my world on Fool Craft and don't know how to work the backup.

**Comment by Jo-Mamma75** [comment]
Do you have a "backups" folder inside your instance folder for FoolCraft?

**Comment by ImTheStigHonest** [comment]
in the backup folder there will be the last server backup, it is a compressed zip file. you will need to use winzip or similar to extract it. then move it to your saves folder

**Comment by EroAxe** [comment]
I found the backup file I just wasn't sure how to restore back to it.  Where would I find the saves folder?? And would I bring the whole unzipped file into it?

**Comment by Jo-Mamma75** [comment]
It's been so long since I've played single player I honestly couldn't tell you were the files and folders you need are located.  That kind of makes me a little sad.

**Comment by EroAxe** [comment]
Darn it, because I had actually just started to get some automatic farms going to.  Well at least I know where I have to move the backup now though.  Thanks guys!

**Comment by EroAxe** [comment]
If anyone can tell me where to find the saves folder it would be very appreciated because I can't seem to find it so the world is still gone.  I'm going to try and search to find it but I can't seem to.

**Comment by SS396** [comment]
The saves folder should be inside the folder with the same name as the profile name in the twitch app.  In other words I named my profile "Foolcraft", so inside the Instances folder there is a folder named "FoolCraft" and then "saves" is located in that directory.  you know if you are in the right place when you see folders with the names config, resources, and screenshots.

To restore your backup, you need to extract the contents of the zip file, not copy it.  Inside the zip file is a folder that has the same name as the world you typed in when you created the world in minecraft, you have to extract that folder (and make sure you retain every files path, it should be an option when extracting) to the saves folder, so when you look at the saves folder you will only see a folder for each world.

Just a guess but I think if you didn't delete the saves folder on purpose you might of clicked and dragged it into a different directory on accident, which has the effect of moving the entire subdirectory.  Start looking through every folder for one thats named "saves" and contains a folder with your world name in it.  When you find it just simply move "saves" back to where it should be, in the same folder with config, resources, and screenshots.

**Comment by EroAxe** [comment]
SS396 The thing is before this I had no idea where to even find the saves folder.  So I couldn't have moved it.

**Comment by EroAxe** [comment]
I'm trying what you said SS396 I'll say here if it worked.

**Comment by Jo-Mamma75** [comment]
It's also possible that when you updated the instance and Curse/Twitch you chose the option to "Update to a New Profile".  Do you have two instances of FoolCraft on Curse/Twitch?  Just curious.

**Comment by SS396** [comment]
> SS396 The thing is before this I had no idea where to even find the saves folder. So I couldn't have moved it.

Well, they don't just delete themselves, and if you or someone else deleted the world from inside minecraft it wouldn't of deleted the saves folder.  Its possible it didn't get copied over when you were upgrading to a newer version, but you haven't actually said what you were doing when the issue happened. 

Let me see if I can explain it differently.

Open the twitch app, and go to the minecraft tab.  You should see your profile(s) that you have previously created.

Left click on the center of the Foolcraft logo (chance cube), that will "view the profile".

Wait a little bit for the mods to display, and then when they finish in the upper right corner there are 3 periods (...)  next to "Website" and "get more content"  Left click on the periods.

Then select open folder.

That will open the foolcraft folder for that profile on your computer in windows explorer.

you should see a list of directories with the following names
 
asm
backups
config
...
resourcepacks
saves
screenshots
...
woot

and then a bunch of files

If you see the rest of the folders listed but don't see one named saves, just make one and name it saves.

To restore the backup you extract the zip file to the saves folder located in the main foolcraft folder.  Inside of the saves folder it should only be a folder with the name of the world, and inside of that folder contains all the files for that world.

In other words open the backup zip file in winzip and then click (and hold down) on the folder of your world name, and then drag it over the saves folder in another explorer window and release the mouse button.

When you open the saves folder, you should only see single folder(s) of your world names, if you don't you extracted it incorrectly.


I hope you get it working.
